apps/sel4test-tests/arch/arm/arch/debug.h

```diff
@@ -5,16 +5,36 @@
  */
 #pragma once
 
+#ifdef CONFIG_ARCH_AARCH32
 #define TEST_SOFTWARE_BREAK_ASM() \
     asm volatile( \
         ".global sbreak, post_sbreak\n\t" \
         ".type post_sbreak, function\n\t" \
         "sbreak:\n\t" \
         "bkpt\n\t")
+#else
+#define TEST_SOFTWARE_BREAK_ASM() \
+    asm volatile( \
+        ".global sbreak, post_sbreak\n\t" \
+        ".type post_sbreak, function\n\t" \
+        "sbreak:\n\t" \
+        "brk #0\n\t")
+#endif
+
+#define TEST_SINGLE_STEP_ASM() \
+    asm volatile( \
+        ".global label_one, label_two\n\t" \
+        "label_one: \n\t" \
+        "nop \n\t" \
+        "label_two: \n\t" \
+        "nop \n\t" \
+        )
 
 /* Tell C about the symbols exported by the ASM above. */
-extern char sbreak;
+extern char sbreak, label_one, label_two;
 #define TEST_SOFTWARE_BREAK_EXPECTED_FAULT_LABEL sbreak
+#define TEST_SS_LABEL_ONE label_one
+#define TEST_SS_LABEL_TWO label_two
```

Comment:
Not sure why this indirection is done, but anyway, I would call it `TEST_SS_BREAKPOINT1` and 2.